sormas-backend/src/main/java/de/symeda/sormas/backend/action/ActionFacadeEjb.java

```diff
@@ -0,0 +1,199 @@
+
+	@PersistenceContext(unitName = ModelConstants.PERSISTENCE_UNIT_NAME)
+	private EntityManager em;
+
+	@EJB
+	private ActionService actionService;
+	@EJB
+	private UserService userService;
+	@EJB
+	private EventService eventService;
+
+	public Action fromDto(ActionDto source) {
+
+		if (source == null) {
+			return null;
+		}
+
+		boolean creation = false;
+		Action target = actionService.getByUuid(source.getUuid());
+		if (target == null) {
+			creation = true;
+			target = new Action();
+			target.setUuid(source.getUuid());
+			if (source.getCreationDate() != null) {
+				target.setCreationDate(new Timestamp(source.getCreationDate().getTime()));
+			}
+		} else {
+			target.setReplyingUser(userService.getCurrentUser());
+		}
+		DtoHelper.validateDto(source, target);
+
+		target.setReply(source.getReply());
+		target.setCreatorUser(userService.getByReferenceDto(source.getCreatorUser()));
+		target.setDescription(source.getDescription());
+		target.setPriority(source.getPriority());
+		target.setDate(source.getDate());
+		if (target.getActionStatus() != source.getActionStatus() && !creation) {
+			target.setStatusChangeDate(new Date());
+		} else {
+			target.setStatusChangeDate(source.getStatusChangeDate());
+		}
+		target.setActionStatus(source.getActionStatus());
+
+		target.setActionContext(source.getActionContext());
+		if (source.getActionContext() != null) {
+			switch (source.getActionContext()) {
+			case EVENT:
+				target.setEvent(eventService.getByReferenceDto(source.getEvent()));
+				break;
+			default:
+				throw new UnsupportedOperationException(source.getActionContext() + " is not implemented");
+			}
+		} else {
+			target.setEvent(null);
+		}
+
+		return target;
+	}
+
+	public ActionDto toDto(Action action) {
+
+		if (action == null) {
+			return null;
+		}
+
+		ActionDto target = new ActionDto();
+		Action source = action;
+
+		target.setCreationDate(source.getCreationDate());
+		target.setChangeDate(source.getChangeDate());
+		target.setUuid(source.getUuid());
+
+		target.setCreatorUser(UserFacadeEjb.toReferenceDto(source.getCreatorUser()));
+		target.setDescription(source.getDescription());
+		target.setReply(source.getReply());
+		target.setReplyingUser(UserFacadeEjb.toReferenceDto(source.getReplyingUser()));
+		target.setPriority(source.getPriority());
+		target.setDate(source.getDate());
+		target.setStatusChangeDate(source.getStatusChangeDate());
+		target.setActionContext(source.getActionContext());
+		target.setActionStatus(source.getActionStatus());
+		target.setEvent(EventFacadeEjb.toReferenceDto(source.getEvent()));
+
+		return target;
+	}
+
+	@Override
+	public ActionDto saveAction(ActionDto dto) {
+
+		Action ado = fromDto(dto);
+		actionService.ensurePersisted(ado);
+		return toDto(ado);
+	}
+
+	@Override
+	public ActionDto getByUuid(String uuid) {
+		return toDto(actionService.getByUuid(uuid));
+	}
+
+	@Override
+	public void deleteAction(ActionDto actionDto) {
+
+		if (!userService.hasRole(UserRole.ADMIN)) {
```

Comment:
Please validate based on user rights instead of user roles

Reply:
OK